lib/ruby_lsp/requests/selection_ranges.rb

```diff
@@ -72,33 +68,64 @@ def initialize(document)
 
       sig { override.returns(T.all(T::Array[Support::SelectionRange], Object)) }
       def run
-        visit(@document.tree) if @document.parsed?
+        visit(@document.tree)
         @ranges.reverse!
       end
 
       private
 
-      sig { override.params(node: T.nilable(SyntaxTree::Node)).void }
+      sig { override.params(node: T.nilable(YARP::Node)).void }
       def visit(node)
         return if node.nil?
 
-        range = create_selection_range(node.location, @stack.last)
-
+        range = if node.is_a?(YARP::InterpolatedStringNode)
+          create_heredoc_selection_range(node, @stack.last)
+        else
+          create_selection_range(node.location, @stack.last)
+        end
         @ranges << range
+
         return if node.child_nodes.empty?
 
-        @stack << range if NODES_THAT_CAN_BE_PARENTS.include?(node.class)
+        if NODES_THAT_CAN_BE_PARENTS.include?(node.class)
+          @stack << range
+        end
```

Comment:
NIT: let's go back to the previous style.
```suggestion
        @stack << range if NODES_THAT_CAN_BE_PARENTS.include?(node.class)
```